src/planner/plan/append_extend.cpp

```diff
@@ -115,41 +115,28 @@ void QueryPlanner::appendNonRecursiveExtend(std::shared_ptr<NodeExpression> boun
 void QueryPlanner::appendRDFPredicateIRIOffsetHashJoin(
     std::shared_ptr<kuzu::planner::LogicalExtend> rdfExtend, kuzu::planner::LogicalPlan& plan) {
     auto relExpression = rdfExtend->getRel();
-    auto nodeExpression = rdfExtend->getNbrNode();
-
-    // We need to add the following properties to the hash join
-    auto rdfNodeIRIProperty =
-        nodeExpression->getPropertyExpression(common::InternalKeyword::RDF_IRI_PROPERTY_NAME);
-    auto rdfRelIRIProperty =
-        relExpression->getPropertyExpression(common::InternalKeyword::RDF_IRI_PROPERTY_NAME);
-    auto rdfNodeInternalIDProperty = nodeExpression->getInternalIDProperty();
-    auto rdfPredicateIRIOffsetProperty = relExpression->getPropertyExpression(
+    auto nodeExpression = binder.createQueryNode(relExpression->getUniqueName(),
```

Comment:
So this is where major change happens. I understand what you are doing here. This logic should be done during binding (let me do it, no need to change this PR).

The general design is each module doesn't not expose to each other. Instead the communicate through some protocol. e.g.
```
Parser -> Statement -> Binder
Binder -> BoundStatement -> Planner
Planner -> LogicalPlan -> Processor
Processor -> PhysicalPlan -> Task Schedular 
```

We should always enforce these constraints as more features being added.

I'll modify this and u can review my PR later on.